src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -534,19 +534,19 @@ public static string GetFormMethodString(FormMethod method)
         /// <inheritdoc />
         public HtmlString TextArea(string expression, string value, int rows, int columns, object htmlAttributes)
         {
-            var metadata = ExpressionMetadataProvider.FromStringExpression(expression, ViewData, MetadataProvider);
+            var modelExplorer = ExpressionMetadataProvider.FromStringExpression(expression, ViewData, MetadataProvider);
             if (value != null)
             {
-                metadata.Model = value;
+                modelExplorer = modelExplorer.GetExplorerForExpression(typeof(string), value);
```

Comment:
on second thought, this metadata may be incorrect if the passed `expression` references a property with implicit conversions to / from a string.  probably safer to use `modelExplorer.ModelMetadata` if `!modelExplorer.ModelMetadata.IsComplexType`.